VoodooRMI/RMIBus.cpp

```diff
@@ -230,24 +230,33 @@ int RMIBus::rmi_register_function(rmi_function *fn) {
     RMIFunction * function;
     
     switch(fn->fd.function_number) {
-        case 0x01:
+        case 0x01: /* device control */
             function = OSDynamicCast(RMIFunction, OSTypeAlloc(F01));
             break;
-        case 0x03:
+        case 0x03: /* PS/2 pass-through */
             function = OSDynamicCast(RMIFunction, OSTypeAlloc(F03));
             break;
-        case 0x11:
+        case 0x11: /* multifinger pointing */
             function = OSDynamicCast(RMIFunction, OSTypeAlloc(F11));
             break;
-        case 0x12:
+        case 0x12: /* multifinger pointing */
             function = OSDynamicCast(RMIFunction, OSTypeAlloc(F12));
             break;
-        case 0x30:
+        case 0x30: /* GPIO and LED controls */
             function = OSDynamicCast(RMIFunction, OSTypeAlloc(F30));
             break;
+//        case 0x09: /* self test (aka BIST) */
+//        case 0x17: /* pointing sticks */
+//        case 0x19: /* capacitive buttons */
+//        case 0x1A: /* simple capacitive buttons */
+//        case 0x21: /* force sensing */
+//        case 0x32: /* timer */
```

Comment:
Thanks for documenting all these, I was wondering what they were - what is your source for these if you don't mind me asking?

Reply:
iirc they are from [official doc](http://www.synaptics.com/sites/default/files/511-000136-01-Rev-E-RMI4-Interfacing-Guide.pdf) (not available now but still available somewhere), [rmi4/Kconfig](https://github.com/torvalds/linux/blob/master/drivers/input/rmi4/Kconfig) and [mailing list](https://lwn.net/Articles/512237/)